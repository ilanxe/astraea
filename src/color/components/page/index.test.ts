import {actions, data} from '.'

describe('<ColorPage>', () => {
  it('should inject data from store as expected', () => {
    const store = {
      color: 'red',
      other: 'else',
    }
    expect(data(store, {})).toMatchSnapshot()
  })

  it('should inject actions as expected', () => {
    const dispatch = jest.fn()
    const {onPreviousColor, onNextColor} = actions(dispatch)
    onPreviousColor()
    onNextColor()
    expect(dispatch.mock.calls).toMatchSnapshot()
  })
})
