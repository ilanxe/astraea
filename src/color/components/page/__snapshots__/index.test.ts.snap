// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorPage> should inject actions as expected 1`] = `
Array [
  Array [
    Object {
      "type": "COLORS_PREVIOUS_COLOR",
    },
  ],
  Array [
    Object {
      "type": "COLORS_NEXT_COLOR",
    },
  ],
]
`;

exports[`<ColorPage> should inject props as expected 1`] = `
Object {
  "color": "red",
}
`;
